web_poet/fields.py

```diff
@@ -81,54 +83,77 @@ def __init__(self, method):
                     f"@field decorator must be used on methods, {method!r} is decorated instead"
                 )
             self.original_method = method
-            self.unbound_method = None
-            self.processors: List[Tuple[Callable, bool]] = []
+            self.name: Optional[str] = None
 
         def __set_name__(self, owner, name):
+            self.name = name
             if not hasattr(owner, _FIELDS_INFO_ATTRIBUTE_WRITE):
                 setattr(owner, _FIELDS_INFO_ATTRIBUTE_WRITE, {})
 
             field_info = FieldInfo(name=name, meta=meta, out=out)
             getattr(owner, _FIELDS_INFO_ATTRIBUTE_WRITE)[name] = field_info
 
         def __get__(self, instance, owner=None):
-            if self.unbound_method is None:
-                for processor in out or []:
-                    sig = inspect.signature(processor)
-                    self.processors.append((processor, "page" in sig.parameters))
-                method = self._processed(self.original_method, instance)
-                if cached:
-                    self.unbound_method = cached_method(method)
+            # We use the original method and the out arg from the field and
+            # the Processors class from the instance class, so caching needs to
+            # take into account the instance class and the field object. So we
+            # use the field object id() as a key when caching the method in
+            # the instance class.
+            cache_key = id(self)
+            method = self._get_processed_method(owner, cache_key)
+            if method is None:
+                if out is not None:
+                    processor_methods = out
+                elif hasattr(instance, "Processors"):
+                    processor_methods = getattr(owner.Processors, self.name, [])
```

Comment:
Is there a difference between instance.Processors and owner.Processors? I'm asking because the code checks that instance has "Processors" attribute, but then acesses owner.Processors. 

If there is no difference, it'd be probably better to use the same approach in both cases. If there is a difference, could you please add a comment to the code which explains it?

Reply:
> I'm asking because the code checks that instance has "Processors" attribute, but then acesses owner.Processors.

Good catch, this is indeed a mistake, but which of them we use is not that important, unless the user changes the instance's `Processors` attribute.